### PR TITLE
JS: Use the new non-extending-subtypes syntax

### PR DESCRIPTION
**Pull request by erik-krogh**
I grepped through the JavaScript QL for `::Range`, and first I converted them all to the casting-based `::Range` pattern, and next to the new syntax.  
I think I got most, but I might have missed some. 

[Evaluation looks fine.](https://github.com/github/codeql-dca-main/tree/run/erik-krogh/casting-nightly-security/reports)   
(No noticeable performance change, no change in results). 

/cc @ginsbach

### CODE REVIEW

File: javascript/ql/lib/semmle/javascript/frameworks/NodeJSLib.qll [hunk 1]
```diff
@@ -826,7 +826,7 @@ module NodeJSLib {
    * for example `http.request(url)`.
    */
   class NodeJSClientRequest extends ClientRequest {
-    override NodeJSClientRequest::Range self;
+    NodeJSClientRequest() { this instanceof NodeJSClientRequest::Range }
```
Comment (asgerf) [inline javascript/ql/lib/semmle/javascript/frameworks/NodeJSLib.qll:829]: Is there a reason we can't use `instanceof` in the base type here?

**Review by asgerf — DISMISSED** [review]
LGTM

File: javascript/ql/lib/semmle/javascript/frameworks/NodeJSLib.qll (hunk 1 above)
Comment (erik-krogh) [reply javascript/ql/lib/semmle/javascript/frameworks/NodeJSLib.qll]: I suppose not.  
`ClientRequest` is already instance of the `::Range` pattern, so I suppose I didn't want to introduce "nested" `::Range` patterns.  
But it should just work, so I've changed it.

**Review by asgerf — APPROVED** [review]

**Comment by intrigus-lgtm** [comment]
Is this comment (and similar ones) still correct?
> * New base64 encoding functions can be supported by **extending** this class.

https://github.com/erik-krogh/ql/blob/a756ffa3a6c388220eb93a1951de68c7fa08f7e3/javascript/ql/lib/semmle/javascript/Base64.qll#L21

**Comment by erik-krogh** [comment]
> Is this comment (and similar ones) still correct?

Yes.

**Pull request merged**